### PR TITLE
implemented a cp function

### LINKED ISSUE #119: vault cp
vault mv copies and deletes. We don't have anything that copies without deleting.

It's just about making this line optional:
https://github.com/peopledoc/vault-cli/blob/5af979a554304bc34b0bf10c039491f8dc2a0433/vault_cli/client.py#L388

### PR DESCRIPTION
**Pull request by soualid**
Closes #119 

### Successful PR Checklist:

- [x] Tests (units, integration, 100% coverage)
- [x] Documentation
- [x] Had a good time contributing?

---

Needed this option, ran into your utility, and discovered it was missing. 

It's almost the first time I code something using python, don't hesitate to tell me if something is wrong! :slightly_smiling_face:

### CODE REVIEW

File: vault_cli/client.py [hunk 1]
```diff
@@ -441,7 +441,13 @@ def move_secrets_iter(
...
             secret_ = cast(types.JSONDict, secret)
             self.set_secret(new_path, secret_, force=force)
-            self.delete_secret(old_path)
+            if delete_source:
+                self.delete_secret(old_path)
+
+    def move_secrets_iter(
+        self, source: str, dest: str, force: Optional[bool] = None
+    ) -> Iterable[Tuple[str, str]]:
+        return self.copy_secrets_iter(source, dest, force, True)
```
Comment (elemoine) [inline vault_cli/client.py:450]: For more clarity:

```suggestion
        return self.copy_secrets_iter(source, dest, force, delete_source=True)
```

**Review by elemoine — APPROVED** [review]
Minor comment, otherwise looks good to me!

File: vault_cli/client.py (hunk 1 above)
Comment (soualid) [reply vault_cli/client.py]: Sure, just changed that!

**Review by ewjoachim — APPROVED** [review]
Hello, welcome to the repository, and thanks for your first contribution!

This looks absolutely awesome. I think we have a slight configuration issue keeping the CI to run on your code, but let's fix that :)

I believe we can get this merged & released soon !

**Comment by ewjoachim** [comment]
Can I ask you to rebase on master or to merge master into your branch ?

**Comment by soualid** [comment]
Sure

**Comment by soualid** [comment]
Rebased and pushed. :ok_hand:

**Comment by ewjoachim** [comment]
Looks like there's a very slight hole in coverage: `copy_secrets` is never called with `generator=False`

**Comment by soualid** [comment]
Sure, I was missing the client tests, which are using this `generator` parameter. I just implemented them and we should be good by now! Cheers.

**Comment by ewjoachim** [comment]
Well I can only say congratulations :) And you said it was one of your first Python PRs, I'm really impressed !

I'm going to make a release :)

**Comment by ewjoachim** [comment]
https://pypi.org/project/vault-cli/

**Pull request merged**